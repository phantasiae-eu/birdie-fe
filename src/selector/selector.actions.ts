import { DbColumns } from './selector.model'
import { ActionCreator } from 'redux'

export const CHANGE_SELECTOR = 'CHANGE_SELECTOR'
export const CHANGE_SELECTOR_ACCEPTED = 'CHANGE_SELECTOR_ACCEPTED'
export const CHANGE_SELECTOR_REJECTED = 'CHANGE_SELECTOR_REJECTED'

export interface AChangeSelector {
    selector: DbColumns
    type: typeof CHANGE_SELECTOR
}

export interface AChangeSelectorAccepted {
    selector: DbColumns
    type: typeof CHANGE_SELECTOR_ACCEPTED
}

export interface AChangeSelectorRejected {
    err: Error
    type: typeof CHANGE_SELECTOR_REJECTED
}

export const changeSelector: ActionCreator<AChangeSelector> = (
    selector: DbColumns
): AChangeSelector => ({
    selector,
    type: CHANGE_SELECTOR,
})

export const changeSelectorAccepted: ActionCreator<AChangeSelectorAccepted> = (
    selector
): AChangeSelectorAccepted => ({
    selector,
    type: CHANGE_SELECTOR_ACCEPTED,
})

export const changeSelectorRejected: ActionCreator<AChangeSelectorRejected> = (
    err: Error
): AChangeSelectorRejected => ({
    err,
    type: CHANGE_SELECTOR_REJECTED,
})
