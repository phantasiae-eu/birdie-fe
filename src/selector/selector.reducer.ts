import { Reducer } from 'redux'
import {
    SelectorStateReducer,
    defaultSelectorStateReducer,
} from './selector.model'
import {
    AChangeSelectorAccepted,
    CHANGE_SELECTOR_ACCEPTED,
} from './selector.actions'

export const selector: Reducer<
    SelectorStateReducer,
    AChangeSelectorAccepted
> = (
    state: SelectorStateReducer = defaultSelectorStateReducer,
    action: AChangeSelectorAccepted
): SelectorStateReducer => {
    switch (action.type) {
        case CHANGE_SELECTOR_ACCEPTED: {
            return { selector: action.selector }
        }

        default:
            return state
    }
}
