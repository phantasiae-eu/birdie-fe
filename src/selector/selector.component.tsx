import React, { ReactElement } from 'react'
import {
    SelectorStateProps,
    SelectorDispatchProps,
    DbItems,
} from './selector.model'
import { Dropdown, SafeAnchor } from 'react-bootstrap'
import { AChangeSelector } from './selector.actions'
import { ReplaceProps, BsPrefixProps } from 'react-bootstrap/helpers'
import { DropdownItemProps } from 'react-bootstrap/DropdownItem'

const selector: React.FC<SelectorStateProps & SelectorDispatchProps> = (
    props: SelectorStateProps & SelectorDispatchProps
): ReactElement => (
    <Dropdown>
        <Dropdown.Toggle variant="success" id="dropdown-basic">
            {
                DbItems.filter(
                    (item: any): any => item.value === props.selector
                )[0].key
            }
        </Dropdown.Toggle>

        <Dropdown.Menu>
            {DbItems.map((item: any, index: number): any => (
                <Dropdown.Item
                    key={index}
                    onClick={(
                        e: React.MouseEvent<
                            ReplaceProps<
                                typeof SafeAnchor,
                                BsPrefixProps<typeof SafeAnchor> &
                                    DropdownItemProps
                            >,
                            MouseEvent
                        >
                    ): void => e.preventDefault()}
                    onSelect={(eventKey: any): AChangeSelector =>
                        props.changeSelector(eventKey)
                    }
                    href={item.value}
                >
                    {item.key}
                </Dropdown.Item>
            ))}
        </Dropdown.Menu>
    </Dropdown>
)

export default selector
