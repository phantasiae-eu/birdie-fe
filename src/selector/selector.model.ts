import { AChangeSelector } from './selector.actions'
import { Dispatch } from 'redux'

export enum DbColumns {
    classOfWorker = 'class%20of%20worker',
    industryCode = 'industry%20code',
    occupationCode = 'occupation%20code',
    education = 'education',
    wagePerHour = 'wage%20per%20hour',
    lastEducation = 'last%20education',
    maritalStatus = 'marital%20status',
    majorIndustryCode = 'major%20industry%20code',
    majorOccupationCode = 'major%20occupation%20code',
    mace = 'mace',
    hispanice = 'hispanice',
    sex = 'sex',
    memberOfLabor = 'member%20of%20labor',
    reasonForUnemployment = 'reason%20for%20unemployment',
    fulltime = 'fulltime',
    capitalGain = 'capital%20gain',
    capitalLoss = 'capital%20loss',
    dividends = 'dividends',
    incomeTaxLiability = 'income%20tax%20liability',
    previousResidenceRegion = 'previous%20residence%20region',
    previousResidenceState = 'previous%20residence%20state',
    householdWithFamily = 'household-with-family',
    householdSimple = 'household-simple',
    weight = 'weight',
    msaChange = 'msa-change',
    regChange = 'reg-change',
    withinRegChange = 'within-reg-change',
    livedHere = 'lived-here',
    migrationPrevResInSunbelt = 'migration%20prev%20res%20in%20sunbelt',
    w = 'w',
    familyMembersUnder20118 = 'family%20members%20under%20118',
    fatherBirthCountry = 'father%20birth%20country',
    motherBirthCountry = 'mother%20birth%20country',
    birthCountry = 'birth%20country',
    citizenship = 'citizenship',
    ownBusinessOrSelfEmployed = 'own%20business%20or%20self%20employed',
    fillQuestionnaireForVeteranSAdmin = 'fill%20questionnaire%20for%20veteran%20s%20admin',
    veteransBenefits = 'veterans%20benefits',
    weeksWorkedInYear = 'weeks%20worked%20in%20year',
    year = 'year',
    salaryRange = 'salary%20range',
}

export const DbItems: any[] = Object.keys(DbColumns).map((key: any) => ({
    key,
    value: DbColumns[key],
}))

export interface SelectorStateReducer {
    selector: DbColumns
}

export const defaultSelectorStateReducer: SelectorStateReducer = {
    selector: DbColumns.classOfWorker,
}

export interface SelectorStateProps {
    selector: DbColumns
}

export interface SelectorDispatchProps {
    changeSelector: (selector: DbColumns) => AChangeSelector
    dispatch: Dispatch
}
