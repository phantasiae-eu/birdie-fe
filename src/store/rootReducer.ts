import { combineReducers } from 'redux'
import { selector } from '../selector/selector.reducer'

const rootReducer = combineReducers({ selector })
export default rootReducer
