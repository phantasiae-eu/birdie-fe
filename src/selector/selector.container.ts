import { connect } from 'react-redux'
import Selector from './selector.component'
import { AppState } from '../store/store.model'
import {
    SelectorStateProps,
    SelectorDispatchProps,
    DbColumns,
} from '../selector/selector.model'
import { Dispatch } from 'redux'
import { changeSelector, AChangeSelector } from './selector.actions'

const mapStateToProps = (state: AppState): SelectorStateProps => ({
    selector: state.selector.selector,
})

const mapDispatchToProps = (dispatch: Dispatch): SelectorDispatchProps => ({
    changeSelector: (selector: DbColumns): AChangeSelector =>
        dispatch(changeSelector(selector)),
    dispatch,
})

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(Selector)
