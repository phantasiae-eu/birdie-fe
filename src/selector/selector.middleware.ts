import { Middleware, Dispatch } from 'redux'
import { AppState } from '../store/store.model'
import {
    AChangeSelector,
    CHANGE_SELECTOR,
    changeSelectorRejected,
} from './selector.actions'

export const selectorMiddleware: Middleware<{}, AppState> = (
    store
): ((
    next: Dispatch<AChangeSelector>
) => (action: AChangeSelector) => Promise<AChangeSelector>) => (
    next: Dispatch<AChangeSelector>
): ((action: AChangeSelector) => Promise<AChangeSelector>) => async (
    action: AChangeSelector
): Promise<AChangeSelector> => {
    const result = next(action)
    switch (action.type) {
        case CHANGE_SELECTOR:
            try {
            } catch (err) {
                store.dispatch(changeSelectorRejected(err))
            }
            break

        default:
            break
    }
    return result
}
