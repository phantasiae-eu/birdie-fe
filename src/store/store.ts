import { createStore, applyMiddleware } from 'redux'
import rootReducer from './rootReducer'
import { composeWithDevTools } from 'redux-devtools-extension'
import { selectorMiddleware } from '../selector/selector.middleware'

const store = createStore(
    rootReducer,
    composeWithDevTools(applyMiddleware(selectorMiddleware))
)

export default store
