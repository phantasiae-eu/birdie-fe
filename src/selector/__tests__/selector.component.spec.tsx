import React from 'react'
import Selector from '../selector.component'
import renderer from 'react-test-renderer'
import { CHANGE_SELECTOR, AChangeSelector } from '../selector.actions'
import { DbColumns } from '../selector.model'
import store from '../../store/store'
describe('selector.component', (): void => {
    it('renders without crashing', (): void => {
        const tree = renderer
            .create(
                <Selector
                    dispatch={store.dispatch}
                    selector={undefined}
                    changeSelector={(): AChangeSelector => ({
                        selector: DbColumns.birthCountry,
                        type: CHANGE_SELECTOR,
                    })}
                />
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})

export default undefined
