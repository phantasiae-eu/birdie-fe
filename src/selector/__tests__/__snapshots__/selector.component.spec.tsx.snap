// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selector.component renders without crashing 1`] = `
<div
  className="dropdown"
  onKeyDown={[Function]}
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    className="dropdown-toggle btn btn-success"
    disabled={false}
    id="dropdown-basic"
    onClick={[Function]}
    type="button"
  >
    Select please
  </button>
</div>
`;
